### PR TITLE
Change max password length to at least 64 characters

### PR DESCRIPTION
**Pull request by sohsatoh**
Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [x] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [x] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [x] Any references to websites have been formatted as `[TEXT](URL)`
- [x] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [x] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #1376 

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Authentication_Cheat_Sheet.md [hunk 1]
```diff
@@ -304,7 +304,7 @@ Web applications should not make the job of password managers more difficult tha
 
 - Use standard HTML forms for username and password input with appropriate `type` attributes.
 - Avoid plugin-based login pages (such as Flash or Silverlight).
-- Implement a reasonable maximum password length, such as 64 characters, as discussed in the [Password Storage Cheat Sheet](Password_Storage_Cheat_Sheet.md#maximum-password-lengths).
+- Implement a reasonable maximum password length, at least 64 characters, as discussed in the [Implement Proper Password Strength Controls below](#implement-proper-password-strength-controls).
```
Comment (szh) [inline cheatsheets/Authentication_Cheat_Sheet.md:307]: Did you mean "above"? I think just "section" is fine.
```suggestion
- Implement a reasonable maximum password length, at least 64 characters, as discussed in the [Implement Proper Password Strength Controls section](#implement-proper-password-strength-controls).
```

**Review by szh — COMMENTED** [review]
I think this is a good improvement. I left one nit.

**Review by kwwall — APPROVED** [review]
LGTM. This reads much better now IMO.

File: cheatsheets/Authentication_Cheat_Sheet.md (hunk 1 above)
Comment (sohsatoh) [reply cheatsheets/Authentication_Cheat_Sheet.md]: Thanks, I just fixed it!

**Review by szh — APPROVED** [review]
LGTM!

**Review by jmanico — APPROVED** [review]

**Pull request merged**